dll/shellext/zipfldr/CZipExtract.cpp

```diff
@@ -591,10 +593,14 @@ class CZipExtract :
 
             bool is_dir = Name.GetLength() > 0 && Name[Name.GetLength()-1] == '/';
 
-            char CombinedPath[MAX_PATH * 2] = { 0 };
-            PathCombineA(CombinedPath, BaseDirectory, Name);
-            CStringA FullPath = CombinedPath;
-            FullPath.Replace('/', '\\');    /* SHPathPrepareForWriteA does not handle '/' */
+            // Build a combined path
+            CPathW CombinedPath;
+            CombinedPath.Combine(BaseDirectory, Name);
+
+            // SHPathPrepareForWrite does not handle '/', even on MS Windows
+            CStringW& FullPath = CombinedPath.m_strPath;
+            FullPath.Replace(L'/', L'\\');
```

Comment:
```suggestion
            CPathW CombinedPath = BaseDirectory;
            BaseDirectory += Name;

            // SHPathPrepareForWrite does not handle '/', even on MS Windows
            CombinedPath.m_strPath.Replace(L'/', L'\\');
```

Reply:
Tbh I don't see the point of `, even on MS Windows` but eh...

Reply:
```c
            CStringW strText;
            strText.Format(L"Name:%s, BaseDirectory:%s", (LPCWSTR)Name, (LPCWSTR)BaseDirectory);
            ::MessageBoxW(NULL, strText, L"Info", 0);
```
![after](https://github.com/reactos/reactos/assets/2107452/e7d9115a-6a26-4eb0-9110-b61c224815a8)
You can see the variable `Name` can contains forward slashes.

Reply:
Yes, I understand that.
I don't see the point of adding `, even on MS Windows` in the comment

Reply:
I don't insist on that change, but the question popped up in my head how Windows would behave.
I just assumed others might come up with the same question.

With the comment it would be more obvious, that we are not working around a ros-bug here within our own SHPathPrepareForWrite()
And it implies: that it's not worth looking closer whether eliminating that string transformation should be subject of future optimization efforts. That was the only motivation.

If you don't like it, I won't kill anybody for not adding the comment, promised.

Reply:
In this case, `SHPathPrepareForWrite` will create the directories recursively unless any for preparation. The forward slashes are unacceptable in this function.

Reply:
Comments improved.